### PR TITLE
[AC-10: Feature] Refactor register feature

### PR DESCRIPTION
**Pull request by turnkevin**
### Provide description as to what you have done in this pr.

Create signup page and do the localisation. 

### Provide screenshots or videos that the function is working.
image
### Provide screenshots of no warning, error in console.
image

#### When you finished, you will want to make sure you have...

- [x] No merge conflict with `develop` branch.
- [x] Check PR title is correct, (i.e `[AC-xx: Feature] Create pull request template`).
- [x] Check your app will run without crashes.
- [x] Check if there's hard coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] No unnecessary spaces between lines of code.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Check you named your code properly.
- [x] Remove unused comments.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "jira-plugin.workingProject": ""
```
Comment (NorrisWu0) [inline .vscode/settings.json:2]: What is this?

File: .env.development [hunk 2]
```diff
@@ -0,0 +1,4 @@
+NEXT_PUBLIC_GRAPHQL_ENDPOINT=http://a-comosus.com:3100/graphql/
```
Comment (NorrisWu0) [inline .env.development:1]: I don't think we need to commit this, but it can stay here for now.

File: public/locales/cn/auth.json [hunk 3]
```diff
@@ -21,5 +21,41 @@
...
+        "required": "请输入邮箱信息"
+      }
+    },
+    "password": {
+      "placeholder": "密码",
+      "error": {
+        "required": "你不要密码吗？"
+      }
+    },
+    "error": "要嘛不是用户名不对，就是密码不对。",
```
Comment (NorrisWu0) [inline public/locales/cn/auth.json]: I don't think you need this error here.

File: public/locales/cn/auth.json [hunk 4]
```diff
@@ -21,5 +21,41 @@
...
+    "user-exist": "用户已存在，请换用户名。",
```
Comment (NorrisWu0) [inline public/locales/cn/auth.json:46]: Imagine our backend will return different errors. We can group this type of message under `error` object.

```
{
  error: {
    existing-user.
    ...other rule,
  }
}

File: public/locales/cn/auth.json [hunk 5]
```diff
@@ -21,5 +21,41 @@
...
+    "button": "注册",
+    "forget-password": "忘记密码",
+    "no-account": {
+      "description": "没注册吗？",
+      "action": "创建一个吧!"
+    },
+    "policy": {
+      "title": "创建帐户即表示您同意我们的条款和条件。查看我们的隐私声明，了解我们如何管理您的个人信息。",
```
Comment (NorrisWu0) [inline public/locales/cn/auth.json]: u sure this is `title`?

File: src/common/components/generics/Policy.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { FormControl, Checkbox, HStack, Text } from '@chakra-ui/react';
+import { FormErrorMessage } from '@src/common/components';
+import { Controller } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+export default function Policy({ name, control }) {
```
Comment (NorrisWu0) [inline src/common/components/generics/Policy.tsx]: So there are two different scenarios here.

if we are reusing this component else where, should it really be called `Policy`?

If we are not reusing it, then should it be a component under `@common` or does it belongs to its respective module?

File: src/common/components/index.ts [hunk 7]
```diff
@@ -7,7 +7,7 @@ export { default as NavBar } from './generics/NavBar';
 export { default as Footer } from './generics/Footer';
 export { default as SelectLanguage } from './generics/SelectLanguage';
 export { default as ToggleThemeButton } from './generics/ToggleThemeButton';
-
+export { default as Policy } from './generics/Policy';
```
Comment (NorrisWu0) [inline src/common/components/index.ts]: Same as above

File: src/modules/auth/LoginForm/login.gql [hunk 8]
```diff
@@ -7,3 +7,9 @@ mutation Login($payload: LoginDetailInput!) {
     }
   }
 }
+
+mutation Register($payload: RegisterDetailInput!) {
```
Comment (NorrisWu0) [inline src/modules/auth/LoginForm/login.gql]: You can either rename this file to `auth.gql` or create a separate `signup.gql`

File: src/styles/Home.module.css [hunk 9]
```diff
@@ -0,0 +1 @@
+
```
Comment (NorrisWu0) [inline src/styles/Home.module.css]: ?

**Review by NorrisWu0 — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (turnkevin) [reply .vscode/settings.json:2]: I have already cancelled it.

**Review by NorrisWu0 — APPROVED** [review]

**Pull request merged**